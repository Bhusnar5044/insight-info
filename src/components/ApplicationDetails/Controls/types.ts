import { ChangeEvent } from 'react';

export interface IControls {
    onSearch: (value: string) => void;
    onSortChange: (e: ChangeEvent<HTMLSelectElement>) => void;
    sort: string;
}
