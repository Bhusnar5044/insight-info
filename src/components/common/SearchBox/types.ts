export interface InputWrapperProps {
    for?: string;
    fullWidth?: boolean;
}
export interface Props extends InputWrapperProps {
    name?: string;
    type?: string;
    onSearch: (value: string) => void;
    placeholder?: string;
    disabled?: boolean;
    required?: boolean;
}
