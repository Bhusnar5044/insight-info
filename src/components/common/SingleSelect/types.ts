import { ChangeEvent } from 'react';

export type Options = { value: string; label: string; disabled?: boolean }[];

export type Props = {
    options: Options;
    onChange: (e: ChangeEvent<HTMLSelectElement>) => void;
    value: string | number;
};
