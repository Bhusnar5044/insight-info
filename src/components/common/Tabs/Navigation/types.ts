import { ITabNavigation } from '../types';

export interface Props {
    tabs: ITabNavigation[];
    activeTabId: string | number;
    onNavClick: (id: string | number) => void;
}
