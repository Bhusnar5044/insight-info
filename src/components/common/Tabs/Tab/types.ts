import { TabData } from '../types';

export interface Props {
    data: TabData;
}
