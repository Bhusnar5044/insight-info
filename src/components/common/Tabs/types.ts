export interface ITabNavigation {
    id: string | number;
    title: string;
}
export interface TabData extends ITabNavigation {
    content: JSX.Element;
}
export interface Props {
    data: TabData[];
}
