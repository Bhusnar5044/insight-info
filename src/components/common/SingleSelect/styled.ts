import { styled } from '@utils';

export const Select = styled.select`
    background-color: white;
    border-radius: 0.3rem;
    display: inline-block;
    font: inherit;
    line-height: 1.5rem;
    padding: 0.5rem 3.5rem 0.5rem 1rem;
    margin: 0;
    outline: unset;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    -webkit-appearance: none;
    -moz-appearance: none;
    background-image: linear-gradient(45deg, transparent 50%, gray 50%),
        linear-gradient(135deg, gray 50%, transparent 50%), radial-gradient(#ddd 70%, transparent 72%);
    background-position: calc(100% - 20px) calc(1em + 2px), calc(100% - 15px) calc(1em + 2px), calc(100% - 0.5em) 0.5em;
    background-size: 5px 5px, 5px 5px, 1.5em 1.5em;
    background-repeat: no-repeat;

    :focus {
        background-image: linear-gradient(45deg, white 50%, transparent 50%),
            linear-gradient(135deg, transparent 50%, white 50%), radial-gradient(gray 70%, transparent 72%);
        background-position: calc(100% - 15px) 1em, calc(100% - 20px) 1em, calc(100% - 0.5em) 0.5em;
        background-size: 5px 5px, 5px 5px, 1.5em 1.5em;
        background-repeat: no-repeat;
    }
`;

export const Option = styled.option`
    padding: 0.5rem;
`;
