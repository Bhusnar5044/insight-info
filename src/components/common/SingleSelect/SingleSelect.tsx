import { FC, memo } from 'react';
import { Box } from '..';
import { Option, Select } from './styled';
import { Props } from './types';

export const SingleSelect: FC<Props> = memo(({ options, onChange, value }) => {
    return (
        <Box gap="1rem">
            <Select title="single-select" value={value} onChange={onChange}>
                {options.map((option) => {
                    return (
                        <Option value={option.value} key={option.value} disabled={option.disabled}>
                            {option.label}
                        </Option>
                    );
                })}
            </Select>
        </Box>
    );
});

SingleSelect.displayName = 'SingleSelect';
