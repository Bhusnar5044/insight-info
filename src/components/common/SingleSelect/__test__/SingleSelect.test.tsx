import { fireEvent, render, screen } from '@test-utils';
import { SingleSelect } from '../SingleSelect';
import { Props } from '../types';

const onChange = jest.fn();
const defaultProps: Props = {
    options: [
        { value: 'dummy', label: 'Dummy' },
        { value: 'dummy1', label: 'Dummy1' },
    ],
    value: '',
    onChange,
};
const renderer = (props = defaultProps) => render(<SingleSelect {...props} />);

describe('SingleSelect', () => {
    it('should render properly with default props', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should open options on click on filed', () => {
        renderer();
        const element = screen.getByTitle('single-select');
        fireEvent.click(element);
        expect(screen.getByText('Dummy')).toBeInTheDocument();
    });

    it('should call onChange method on option change', () => {
        renderer();
        const element = screen.getByTitle('single-select');
        fireEvent.click(element);
        const option = screen.getByText('Dummy');
        expect(option).toBeInTheDocument();
        fireEvent.change(element, { target: { value: 'dummy' } });
        expect(onChange).toBeCalled();
    });
});
