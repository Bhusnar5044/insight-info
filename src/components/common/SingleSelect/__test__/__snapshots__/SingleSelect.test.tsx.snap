// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSelect should render properly with default props 1`] = `
.c0 {
  display: flex;
  gap: 1rem;
}

.c1 {
  width: 100%;
  min-width: 15ch;
  max-width: 30ch;
  border: 1px solid var(--select-border);
  border-radius: 0.25em;
  padding: 0.25em 0.5em;
  font-size: 1.25rem;
  cursor: pointer;
  line-height: 1.1;
  background-color: #fff;
  background-image: linear-gradient(to top, #f9f9f9, #fff 33%);
  display: grid;
  grid-template-areas: 'select';
  align-items: center;
  position: relative;
}

.c1:after {
  content: '';
  width: 0.8em;
  height: 0.5em;
  background-color: var(--select-arrow);
  clip-path: polygon(100% 0%, 0 0%, 50% 100%);
  grid-area: select;
  justify-self: end;
}

.c1:focus+.focus {
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  border: 2px solid var(--select-focus);
  border-radius: inherit;
}

.c2 {
  padding: 0.5rem;
}

<div>
  <div
    class="c0"
    gap="1rem"
  >
    <label
      for="select2"
    >
      Select2
    </label>
    <select
      class="c1"
      title="single-select"
    >
      <option
        class="c2"
        value="dummy"
      >
        Dummy
      </option>
      <option
        class="c2"
        value="dummy1"
      >
        Dummy1
      </option>
    </select>
  </div>
</div>
`;
